Automattic/jetpack
# Carousel: disable WordPress Core' Lightbox option

Fixes #32668

## Proposed changes:

It's best to keep only one lightbox option, to avoid any confusion. Since Jetpack's Carousel feature currently offers more features, let's keep ours in favor of Core's for now.
In the future, when core's lightbox option becomes more robust, we can consider deprecating Jetpack's Carousel feature altogether.

> [!WARNING]
> This currently breaks some tests on WordPress.com, but I'm not sure how to best address that. Should I simply update the tests to expect that data added by Carousel for now?

### Other information:


## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No
## Testing instructions:

> [!NOTE]
> This currently requires the `trunk` version of the Gutenberg plugin. When using the current stable version of the plugin, or no Gutenberg plugin at all, that fix will not work.

* Start with a site using a block-based theme like Twenty Twenty Four
* Go to Jetpack > Settings > Writing and enable the Carousel feature.
* Go to Posts > Add New
* Add a new Image block, and upload an image.
* In the block's toolbar, under the linking options, you should see an option to "expand on click".
* Now switch to this branch.
* Go to Posts > Add New
* Add a new Image block and insert a new image.
* The option should now be gone from the block's toolbar.
* Go to Jetpack > Settings > Writing and disable the Carousel feature.
* Go to Posts > Add New
* Add a new Image block and insert a new image.
* The option should be back in the block toolbar's link settings.
